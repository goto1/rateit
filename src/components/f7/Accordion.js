import React from "react";
import PropTypes from "prop-types";

const getComponent = name => ({ className, children }) => {
  const classNames = `${name} ${className ? className : ""}`.trim();
  return (
    <div className={classNames}>
      {children}
    </div>
  );
};

export const AccordionItem = getComponent("accordion-item");

AccordionItem.propTypes = {
  className: PropTypes.string,
  children: PropTypes.node.isRequired
};

export const AccordionToggle = getComponent("accordion-item-toggle");

AccordionToggle.propTypes = {
  className: PropTypes.string,
  children: PropTypes.node.isRequired
};

export const AccordionContent = getComponent("accordion-item-content");

AccordionContent.propTypes = {
  className: PropTypes.string,
  children: PropTypes.node.isRequired
};

const Accordion = getComponent("accordion-list");

Accordion.propTypes = {
  className: PropTypes.string,
  children: PropTypes.node.isRequired
};

export default Accordion;
