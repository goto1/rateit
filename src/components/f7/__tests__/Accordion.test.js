import React from "react";
import Accordion, {
  AccordionItem,
  AccordionToggle,
  AccordionContent
} from "../Accordion";
import renderer from "react-test-renderer";
import { shallow } from "enzyme";

describe("Accordion component", () => {
  it("should render correctly", () => {
    const tree = renderer
      .create(
        <Accordion>
          <AccordionItem>HelloWorld</AccordionItem>
          <AccordionToggle>HelloWorld</AccordionToggle>
          <AccordionContent>HelloWorld</AccordionContent>
        </Accordion>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("should render AccordionContent component", () => {
    const wrapper = shallow(
      <Accordion>
        <AccordionContent>HelloWorld</AccordionContent>
      </Accordion>
    );
    const actual = wrapper.find(AccordionContent).length;
    const expected = 1;
    expect(actual).toBe(expected);
  });

  it("should render with a custom styling class", () => {
    const wrapper = shallow(
      <Accordion className="custom">
        <span>HelloWorld</span>
      </Accordion>
    );
    const actual = wrapper.props().className.includes("custom");
    const expected = true;
    expect(actual).toBe(expected);
  });

  describe("AccordionItem", () => {
    it("should render a span element", () => {
      const wrapper = shallow(
        <AccordionItem>
          <span>HelloWorld</span>
        </AccordionItem>
      );
      const actual = wrapper.find("span").length;
      const expected = 1;
      expect(actual).toBe(expected);
    });
  });

  describe("AccordionToggle", () => {
    it("should render a span element", () => {
      const wrapper = shallow(
        <AccordionToggle>
          <span>HelloWorld</span>
        </AccordionToggle>
      );
      const actual = wrapper.find("span").length;
      const expected = 1;
      expect(actual).toBe(expected);
    });
  });

  describe("AccordionContent", () => {
    it("should render a span element", () => {
      const wrapper = shallow(
        <AccordionContent>
          <span>HelloWorld</span>
        </AccordionContent>
      );
      const actual = wrapper.find("span").length;
      const expected = 1;
      expect(actual).toBe(expected);
    });
  });
});
