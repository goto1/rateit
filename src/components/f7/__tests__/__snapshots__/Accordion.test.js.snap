// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Accordion component should render correctly 1`] = `


<div
  className="accordion-list"
>
  <div
    className="accordion-item"
  >
    HelloWorld
  </div>
  <div
    className="accordion-item-toggle"
  >
    HelloWorld
  </div>
  <div
    className="accordion-item-content"
  >
    HelloWorld
  </div>
</div>
`;
